sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedFlux.java

```diff
@@ -15,6 +15,8 @@
 import java.util.Collections;
 import java.util.List;
 
+
+// TODO:
```

Comment:
I added this as a note to add tests to this file for v2 as necessary. After reviewing the test cases and the way they are structured, I think it is unnecessary to add further test cases. The logic of the bookkeeping strikes me as the same here even though the sizes of the encryption blocks are different. I am open to other input on this and if I am missing something that indicates I should add explicit v2 tests here

Reply:
a guess. 
how do the numbers below relate to the fact that V2 uses 4MB auth blocks ? would it be interesting to see what happens when "adjustment" happens at the edge of two 4mb blocks ?

Reply:
It should be agnostic of the size. These tests and the calculation in BlobDecryptionPolicy don't use any hard coded numbers/constants. I didn't have to change any of the code around this for v2, only ensure I was feeding this part of the decryption method the right numbers. 

I added some tests that verify we're expanding the range correctly, and those did require specific numbers and caught an error, but I think this code is safe from that. If you have some specific numbers that are of concern, I can either experiment with them or codify them in a test.

Reply:
The test using this class targets the `blobDecryptionPolicy.decryptBlob`. The V1 code "adjustments" seem to be around 16 bytes (same order of magnitude the numbers in EncryptedFlux are).
If we were to apply this testing strategy to V2 then interesting points to test around are 0, 4MB - few bytes, 4MB, 4MB + few bytes and so on (i.e. test buffer staging area and vicinity). 

@jaschrep-msft @gapra-msft could you please chime in and help us decide if/how this test is meaningful for decryption V2 https://github.com/Azure/azure-sdk-for-java/blob/5087bf2173419f89f7e381d267f6d2b8b7de750f/sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/DecryptionTests.groovy#L38 ?

Reply:
I think the test is low-level enough to a specific component that we aren't locked into the 4MB size we are for a broader upload. It looks like we could make encryption v2 operate on the order of 16 bytes for this test. Am I interpreting this correctly @rickle-msft?

It does look like we have encryption tests that are also separated from upload which are using the 4MB size. I'm not sure what the difference is here vs there.

Aside: the test Kamil links to is not very descriptive. It just says "decryption" and it took a while to figure out this was just a range adjustment test (which I'm still not 100% confident on). Can we rename that test to make it clearer what to look for when reading it?